src/progress_bar.rs

```diff
@@ -615,6 +635,26 @@ impl MultiProgress {
         self.join_impl(true)
     }
 
+    /// Modify the state on inserting/adding a progress bar
+    fn extend_state(state: &mut MultiProgressState) -> usize {
+        let object_idx;
+        if let Some(i) = state.free_set.iter().next().cloned() {
+            state.objects[i] = Some(MultiObject {
+                done: false,
+                draw_state: None,
+            });
+            state.free_set.remove(&i);
+            object_idx = i;
+        } else {
+            state.objects.push(Some(MultiObject {
+                done: false,
+                draw_state: None,
+            }));
+            object_idx = state.objects.len() - 1;
+        }
+        object_idx
```

Comment:
Let's extract something like `let new = Some(MultiObject { .. })` from the `if/else` block, and assign the `object_idx` from the block directly (so `let object_idx = if let .. { } else { .. };`)?

Reply:
Sure, will do it.